### PR TITLE
Improve retry mechanism for Pub/Sub to Splunk network errors

### PR DESCRIPTION
**Pull request by melbrodrigues**
- Add IOException handler with exponential backoff for Splunk requests
- Increase read timeout to 60s (previous default value - 20s)

### CODE REVIEW

**Comment by melbrodrigues** [comment]
@rarsan FYI

**Review by rarsan — COMMENTED** [review]
Thx for this PR.
@prathapreddy123 can you review this?
Let's make sure it's the same retry (with exp backoff) behavior that already exists for server 5xx errors when publishing to HEC. Also, are the default retry intervals documented somewhere?

File: src/main/java/com/google/cloud/teleport/splunk/HttpEventPublisher.java [hunk 1]
```diff
@@ -115,10 +119,15 @@ public HttpResponse execute(List<SplunkEvent> events) throws IOException {
...
-
     responseHandler.setBackOffRequired(BackOffRequired.ON_SERVER_ERROR);
-
     request.setUnsuccessfulResponseHandler(responseHandler);
+
+    HttpIOExceptionHandler ioExceptionHandler =
+        new HttpBackOffIOExceptionHandler(getConfiguredBackOff());
+    request.setIOExceptionHandler(ioExceptionHandler);
+
+    request.setReadTimeout(READ_TIMEOUT);
```
[the remark this answers is not in the document]
Comment (melbrodrigues) [reply src/main/java/com/google/cloud/teleport/splunk/HttpEventPublisher.java]: Done

**Comment by melbrodrigues** [comment]
@rarsan default intervals can be found in the documentation [here](https://googleapis.dev/java/google-http-client/latest/index.html?com/google/api/client/util/ExponentialBackOff.Builder.html)

**Review by prathapreddy123 — APPROVED** [review]
LGTM. 

@rarsan - Once you approve I will merge the change

**Comment by rarsan** [comment]
LGTM. Thx!

**Pull request merged**